src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilterElement.java

```diff
@@ -177,7 +177,22 @@ public XpathFilterElement(Pattern files, Pattern checks, Pattern message,
     public boolean accept(TreeWalkerAuditEvent event) {
         return !isFileNameAndModuleAndModuleNameMatching(event)
                 || !isMessageNameMatching(event)
-                || !isXpathQueryMatching(event);
+                || !isXpathQueryMatching(event)
+                || areAllPropertiesNull();
+    }
+
+    /**
+     * Check if all properties are set to null.
+     *
+     * @return true if all properties are set to null
+     */
+
+    private boolean areAllPropertiesNull() {
+        return fileRegexp == null
+                && checkRegexp == null
+                && messageRegexp == null
+                && moduleId == null
+                && xpathExpression == null;
```

Comment:
all these properties are initialized in constructor. So there is no need to check it for every event. You can just setup field in ctor and check this field first in `accept` method.